File: TypeScript/Mit Polymorphismus/Wellensittich.ts
import {Haustier} from "./Haustier";

class Wellensittich extends Haustier{
    constructor(n : string){
        super(n);

    }

    fressen(){
        console.log("Korn fressen");
    }

    freuen(){
        console.log("wippen");
    }

    lautGeben(){
        console.log("zwitschern");
    }
}

export {Wellensittich};

File: C#/Mit Polymorphismus/Wellensittich.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;


class Wellensittich : Haustier
{
    public Wellensittich(string n) : base(n)
    { }
    public override void Fressen()
    {
        System.Console.WriteLine("Korn fressen");
    }

    public override void Freuen()
    {
        System.Console.WriteLine("wippen");
    }

    public override void LautGeben()
    {
        System.Console.WriteLine("zwitschern");
    }
}    

